Assets/Scripts/Inventory/Inventory.cs

```diff
@@ -8,28 +8,37 @@
 /// Static placeholder inventory class for the player. Will probably get moved to something else that makes sense, like the player...or not.
 /// </summary>
 public static class Inventory {
-    public static string[] addedItems = new string[] { };
-    public static int[] addedItemsTypes = new int[] { };
+    public static List<string> addedItems = new List<string>();
+    public static List<int> addedItemsTypes = new List<int>();
     public static LuaInventory luaInventory;
     public static int tempAmount = 0;
-    public static Dictionary<string, string> NametoDesc = new Dictionary<string, string>(), NametoShortName = new Dictionary<string, string>(), NameToReplac = new Dictionary<string, string>();
+    public static Dictionary<string, string> NametoDesc = new Dictionary<string, string>(), NametoShortName = new Dictionary<string, string>();
     public static Dictionary<string, int> NametoType = new Dictionary<string, int>(), NametoPrice = new Dictionary<string, int>();
     public static bool usedItemNoDelete = false;
     //public static bool overworld = false;
     public static List<UnderItem> inventory = new List<UnderItem>();
 
     public static void SetItemList(string[] items = null) {
+        foreach (string item in items) {
+            string desc = "";
+            string sn   = "";
+            int type    =  0;
+            int price   =  0;
+            if (!addedItems.Contains(item) && !NametoDesc.TryGetValue(item, out desc) && !NametoShortName.TryGetValue(item, out sn) && !NametoType.TryGetValue(item, out type) && !NametoPrice.TryGetValue(item, out price))
```

Comment:
Why do you use custom variables? We don't even need them, maybe you should just use some generic variables named "outString" and "outInt" instead of creating a different variable for each different function call.

The suggestion below would replace from line 22 to line 27:

```suggestion
            string outString = "";
            int outInt = 0;
            if (!addedItems.Contains(item) && !NametoDesc.TryGetValue(item, out outString) && !NametoShortName.TryGetValue(item, out outString) && !NametoType.TryGetValue(item, out outInt) && !NametoPrice.TryGetValue(item, out outInt))
```